contracts/HydraDelegation/modules/VestedDelegation/VestedDelegation.sol

```diff
@@ -308,35 +308,34 @@ abstract contract VestedDelegation is
 
         if (reward == 0) return;
 
-        uint256 delegatorReward;
         uint256 stakerCommission;
+        if (position.commission != 0) {
+            (stakerCommission, reward) = _applyCommission(reward, position.commission);
+        }
+
         // Handle additional rewards if the vesting period has ended
         if (block.timestamp >= position.end + position.duration) {
             uint256 baseCommission = delegationCommissionPerStaker[staker];
             uint256 additionalReward = delegationPool.claimRewards(msg.sender);
-            delegatorReward = aprCalculatorContract.applyBaseAPR(additionalReward);
+            additionalReward = aprCalculatorContract.applyBaseAPR(additionalReward);
 
             if (baseCommission != 0) {
-                (stakerCommission, delegatorReward) = _applyCommission(delegatorReward, baseCommission);
+                (uint256 stakerCut, uint256 rewardLeft) = _applyCommission(additionalReward, baseCommission);
```

Comment:
```suggestion
                (uint256 additionalCommission, additional reward) = _applyCommission(additionalReward, baseCommission);
```

Reply:
I am not allowed to declare on of the variables in the return and the other to reuse. But I can declare additionalCommission before the calculation and pass additionalReward!